Sources/TuistKit/Services/PluginArchiveService.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import ProjectDescription
+import TSCBasic
+import TuistDependencies
+import TuistLoader
+import TuistSupport
+
+final class PluginArchiveService {
+    private let swiftPackageManagerController: SwiftPackageManagerControlling
+    private let manifestLoader: ManifestLoading
+    private let fileArchiverFactory: FileArchivingFactorying
+
+    init(
+        swiftPackageManagerController: SwiftPackageManagerControlling = SwiftPackageManagerController(),
+        manifestLoader: ManifestLoading = ManifestLoader(),
+        fileArchiverFactory: FileArchivingFactorying = FileArchivingFactory()
+    ) {
+        self.swiftPackageManagerController = swiftPackageManagerController
+        self.manifestLoader = manifestLoader
+        self.fileArchiverFactory = fileArchiverFactory
+    }
+
+    func run(path: String?) throws {
+        let path = self.path(path)
+
+        let packageInfo = try swiftPackageManagerController.loadPackageInfo(at: path)
+        let taskProducts = packageInfo.products
+            .filter {
+                switch $0.type {
+                case .executable:
+                    return true
+                case .library, .plugin, .test:
+                    return false
+                }
+            }
+            .map(\.name)
+            .filter { $0.hasPrefix("tuist-") }
+
+        if taskProducts.isEmpty {
+            logger.warning("No tasks found - make sure you have executable products with tuist- prefix defined in your manifest.")
+            return
+        }
+
+        let plugin = try manifestLoader.loadPlugin(at: path)
+
+        try FileHandler.shared.inTemporaryDirectory { temporaryDirectory in
+            try archiveProducts(
+                taskProducts: taskProducts,
+                path: path,
+                plugin: plugin,
+                in: temporaryDirectory
+            )
+        }
+    }
+
+    // MARK: - Helpers
+
+    private func path(_ path: String?) -> AbsolutePath {
+        if let path = path {
+            return AbsolutePath(path, relativeTo: FileHandler.shared.currentPath)
+        } else {
+            return FileHandler.shared.currentPath
+        }
+    }
+
+    private func archiveProducts(
+        taskProducts: [String],
+        path: AbsolutePath,
+        plugin: Plugin,
+        in temporaryDirectory: AbsolutePath
+    ) throws {
+        let artifactsPath = temporaryDirectory.appending(component: "artifacts")
+        try taskProducts
+            .forEach { product in
+                try swiftPackageManagerController.buildFatReleaseBinary(
+                    packagePath: path,
+                    product: product,
+                    buildPath: temporaryDirectory.appending(component: "build"),
```

Comment:
Should we put the SPM related constants (e.g. `artifacts`, `build`) in some dedicated file, to be reused here and in `TuistDependencies`?

Reply:
both of those paths are arbitrary, so I'm not sure if there's value in sharing those with `TuistDependencies`?

Reply:
oh right I got confused because they are called the same as some paths created by SPM (e.g. `swift package resolves` downloads artifacts in an `artifacts` folder), but in this case they are not the same :D